include/sys/zio_compress.h

```diff
@@ -107,15 +171,26 @@ extern int lz4_decompress_zfs(void *src, void *dst, size_t s_len, size_t d_len,
     int level);
 extern int lz4_decompress_abd(abd_t *src, void *dst, size_t s_len, size_t d_len,
     int level);
+extern int zstd_init(void);
+extern void zstd_fini(void);
+extern size_t zstd_compress(void *src, void *dst, size_t s_len, size_t d_len,
+    int level);
+extern int zstd_decompress(void *src, void *dst, size_t s_len, size_t d_len,
+    int level);
+extern int zstd_getlevel(void *src, size_t s_len);
```

Comment:
These need to move to the public `include/sys/zstd/zstd.h`.

Reply:
zstd.h belongs to the zstd library. zstd_compress etc. does belong to the zfs specific implementation. so zstd.h is the plain original header of zstd but doesnt belong to zfs in any way. but i will rename zstd_decompress to  zfs_std_decompress to be more specific here